### PR TITLE
 Make requirements with pip-tools 3.8.0 

### PR DESCRIPTION
**Pull request by millerdev**
Attempt to fix travis build failure seen in various PRs (ex: https://github.com/dimagi/commcare-hq/pull/24509).

@emord

### CODE REVIEW

**Review by emord — CHANGES_REQUESTED** [review]
Very confused by this PR.

1. `requests[security]` was changed to just `requests` which made a lot of changes to the requirements on prod.
2. pip-tools is pinned to 3.6.0. To upgrade it to 3.8.0 dev-requirements.in should have chaned and `make upgrade-requirements` should have been run.
3. when i upgraded to 3.8.0 locally I do see the added setup tools, but I don't see all of these other changes that have been made

**Comment by millerdev** [comment]
So one problem I had with the original version of this PR is that I ran `make requirements` with an active Python 3 virtualenv. That's fixed now (force-pushed).

Also, I think this is another problem:

https://github.com/dimagi/commcare-hq/blob/e0fc2748392e04f1261e3211788fd0899eb5a887/scripts/test-make-requirements.sh#L4

Installs the latest version of pip-tools, not the version pinned in `requirements.txt`

File: requirements-python3/dev-requirements.txt [hunk 1]
```diff
@@ -33,7 +33,7 @@ commonmark==0.9.0         # via recommonmark
 concurrent-log-handler==0.9.12
 contextlib2==0.5.5
 coverage==4.5.1
-cryptography==2.6.1
+cryptography==2.7
```
Comment (emord) [inline requirements-python3/dev-requirements.txt]: I'm confused how this was upgraded. `make requirements` is supposed to be strictly making sure that the requirements fit in with current definition in `*.in` files. It should only change when running `make upgrade-requirements` or changing the in files. (it didn't show up in https://github.com/dimagi/commcare-hq/pull/24519/)

File: scripts/test-make-requirements.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-pip install pip-tools==3.6.1
+pip install pip-tools>=3.8
```
Comment (emord) [inline scripts/test-make-requirements.sh]: I think that this is still fragile to this kind of change. Long term, maybe we should run the requirements test after the other tests? that way this comparison could run in the same virtualenv where we have pinned pip-tools.

**Review by emord — APPROVED** [review]
My comment isn't immediately actionable. just an idea. I think this PR is still a better place than it was before

**Review by dannyroberts — APPROVED** [review]

File: scripts/test-make-requirements.sh (hunk 2 above)
Comment (millerdev) [reply scripts/test-make-requirements.sh]: > Long term, maybe we should run the requirements test after the other tests?

Sure, I'll do that in a separate PR

Comment (millerdev) [reply scripts/test-make-requirements.sh]: https://github.com/dimagi/commcare-hq/pull/24522

**Pull request merged**